misk/web/tabs/loader/src/sagas/loaderSaga.ts

```diff
@@ -0,0 +1,102 @@
+// In case you need to use a selector
+// import also select from redux-saga/effects
+// and then simplie yield select(yourSelector())
+//
+// In case you need to redirect to whatever route
+// import { push } from react-router-redux and then
+// yield put(push('/next-page'))
+
+import { IMiskAdminTab, IMiskAdminTabs } from "@misk/common"
+import axios from "axios"
+import { all, call, put, takeEvery, takeLatest } from "redux-saga/effects"
+
+import {
+  dispatchLoader, IAction, IActionType, LOADER
+} from "../actions"
+
+function * handleGetAllComponents (action: IAction<IActionType, { adminTabs: IMiskAdminTabs }>) {
+  try {
+    let adminTabs: any = {}
+    try {
+      const { data } = yield call(axios.get, "http://0.0.0.0:8080/api/admintab/all")
+      adminTabs = data.adminTabs
+      yield put(dispatchLoader.success({ adminTabs }))
+      if (Object.entries(adminTabs).length === 0) {
+        yield put(dispatchLoader.success({ adminTabComponents: { } }))
+      }
+    } catch (e) {
+      yield put(dispatchLoader.failure({ error: { ...e } }))
+    }
+
+    for (const key in adminTabs) {
+      if (adminTabs.hasOwnProperty(key)) {
+        const tab = adminTabs[key]
+        const url = `${tab.url_path_prefix}/tab_${tab.slug}.js`
+        const { data } = yield call(axios.get, url)
+        yield put(dispatchLoader.success({ adminTabComponents: { [tab.slug]: data } }))
+      }
+    }
+  } catch (e) {
+    yield put(dispatchLoader.failure({ error: { ...e } }))
+  }
+}
+
+function * handleGetAllTabs () {
+  try {
+    const { data } = yield call(axios.get, "http://0.0.0.0:8080/api/admintab/all")
+    const { adminTabs } = data
+    yield put(dispatchLoader.success({ adminTabs }))
+  } catch (e) {
+    yield put(dispatchLoader.failure({ error: { ...e } }))
+  }
+}
+
+function * handleGetAllAndTabs (action: IAction<IActionType, {}>) {
+  try {
+    let adminTabs: any = {}
+    try {
+      yield put(dispatchLoader.getAllTabs())
+      const { data } = yield call(axios.get, "http://0.0.0.0:8080/api/admintab/all")
+      adminTabs = data.adminTabs
+      yield put(dispatchLoader.success({ adminTabs }))
+      if (Object.entries(adminTabs).length === 0) {
+        yield put(dispatchLoader.success({ adminTabComponents: { } }))
+      }
+    } catch (e) {
+      yield put(dispatchLoader.failure({ error: { ...e } }))
+    }
+
+    for (const key in adminTabs) {
+      if (adminTabs.hasOwnProperty(key)) {
+        const tab = adminTabs[key]
+        const url = `${tab.url_path_prefix}/tab_${tab.slug}.js`
+        const { data } = yield call(axios.get, url)
+        yield put(dispatchLoader.success({ adminTabComponents: { [tab.slug]: data } }))
+      }
+    }
+  } catch (e) {
```

Comment:
This paradigm feels weird to me. There's a lot of code in the try/catch. I'd like to do a gut check that this is a normal paradigm for saga functions. Is there a control flow issue where throwing an exception ever would be bad? Let's discuss.

Reply:
I would narrow it to something like this: 

```
  try {
    const { data } = yield call(axios.get, url)
    yield put(dispatchLoader.success({ adminTabComponents: { [tab.slug]: data } }))
  } catch (e) {
    yield put(dispatchLoader.failure({ error: { ...e } }))
  }
```